drivers/net/phy/marvell10g_ptp.c

```diff
@@ -348,6 +365,57 @@ int mv3310_ptp_start(struct mv3310_ptp_priv *priv)
 	return ret;
 }
 
+int mv3310_ptp_update(struct mv3310_ptp_priv *priv)
+{
+	int ret;
+
+	if (!priv)
+		return 0;
+
+	mutex_lock(&priv->lock);
+	ret = mv3310_set_mac_mru(priv->phydev, priv->phydev->speed);
+	mutex_unlock(&priv->lock);
+	return 0;
+}
+
+static int mv3310_set_mac_mru(struct phy_device *phydev, int speed)
+{
+	static int old_speed = SPEED_UNKNOWN;
```

Comment:
this will be problematic with multiple instances of the phy (currently does not exists, but it is very possible to have in a future product). I suggest putting it somewhere in the priv context.